core/src/main/java/org/elasticsearch/action/support/replication/ReplicationOperation.java

```diff
@@ -284,6 +283,11 @@ private void decPendingAndFinishIfNeeded() {
 
     private void finish() {
         if (finished.compareAndSet(false, true)) {
+            try {
+                primary.updateGlobalCheckpoint();
```

Comment:
I'm ok with the background sync for this PR. We should in the future also consider the case where the background sync fails. No retry attempt is made. I've left a top-level comment about inlining `updateCheckpointOnPrimary` into `updateLocalCheckpoint`.